includes/class-wc-geolocation.php

```diff
@@ -253,44 +253,32 @@ public static function update_database() {
 			return;
 		}
 
-		require_once ABSPATH . 'wp-admin/includes/file.php';
-
-		$database             = 'GeoLite2-Country.mmdb';
-		$target_database_path = self::get_local_database_path();
-		$tmp_database_path    = download_url( self::GEOLITE2_DB );
-
-		if ( ! is_wp_error( $tmp_database_path ) ) {
-			WP_Filesystem();
-
-			global $wp_filesystem;
-
-			try {
-				// Make sure target dir exists.
-				$wp_filesystem->mkdir( dirname( $target_database_path ) );
+		// There's no need to update the database with no geolocation configured.
+		$license_key = ( new WC_MaxMind_Geolocation_Integration() )->get_option( 'license_key' );
```

Comment:
This keep the dependency on this class, let's change it into a hook, so in the integration we can hook into it and download the file.